pkg/cmd/server/kubernetes/master/controller.go

```diff
@@ -0,0 +1,311 @@
+			VolumePlugins:             plugins,
+			Cloud:                     c.CloudProvider,
+			ClusterName:               ctx.Options.ClusterName,
+			VolumeInformer:            ctx.InformerFactory.Core().V1().PersistentVolumes(),
+			ClaimInformer:             ctx.InformerFactory.Core().V1().PersistentVolumeClaims(),
+			ClassInformer:             ctx.InformerFactory.Storage().V1beta1().StorageClasses(),
+			EventRecorder:             recorder,
+			EnableDynamicProvisioning: ctx.Options.VolumeConfiguration.EnableDynamicProvisioning,
+		})
+
+	go volumeController.Run(ctx.Stop)
+
+	return true, nil
+}
+
+type PersistentVolumeAttachDetachControllerConfig struct {
+	CloudProvider cloudprovider.Interface
+}
+
+func (c *PersistentVolumeAttachDetachControllerConfig) RunController(ctx kubecontroller.ControllerContext) (bool, error) {
+	attachDetachController, err := attachdetachcontroller.NewAttachDetachController(
+		ctx.ClientBuilder.ClientOrDie(bootstrappolicy.InfraPersistentVolumeAttachDetachControllerServiceAccountName),
+		ctx.InformerFactory.Core().V1().Pods(),
+		ctx.InformerFactory.Core().V1().Nodes(),
+		ctx.InformerFactory.Core().V1().PersistentVolumeClaims(),
+		ctx.InformerFactory.Core().V1().PersistentVolumes(),
+		c.CloudProvider,
+		kctrlmgr.ProbeAttachableVolumePlugins(ctx.Options.VolumeConfiguration),
+		ctx.Options.DisableAttachDetachReconcilerSync,
+		ctx.Options.ReconcilerSyncLoopPeriod.Duration,
+	)
+	if err != nil {
+		return false, fmt.Errorf("failed to start attach/detach persistent volume controller: %v", err)
+	}
+
+	go attachDetachController.Run(ctx.Stop)
+
+	return true, nil
+}
+
+type SchedulerControllerConfig struct {
+	PrivilegedClient               kclientset.Interface
+	SchedulerName                  string
+	HardPodAffinitySymmetricWeight int
+	PolicyConfigFile               string
+	SchedulerConfigFile            string
+}
+
+func (c *SchedulerControllerConfig) RunController(ctx kubecontroller.ControllerContext) (bool, error) {
+	var (
+		policy     schedulerapi.Policy
+		configData []byte
+		config     *scheduler.Config
+	)
+
+	// TODO make the rate limiter configurable
+	configFactory := factory.NewConfigFactory(
+		c.SchedulerName,
+		c.PrivilegedClient,
+		ctx.InformerFactory.Core().V1().Nodes(),
+		ctx.InformerFactory.Core().V1().Pods(),
+		ctx.InformerFactory.Core().V1().PersistentVolumes(),
+		ctx.InformerFactory.Core().V1().PersistentVolumeClaims(),
+		ctx.InformerFactory.Core().V1().ReplicationControllers(),
+		ctx.InformerFactory.Extensions().V1beta1().ReplicaSets(),
+		ctx.InformerFactory.Apps().V1beta1().StatefulSets(),
+		ctx.InformerFactory.Core().V1().Services(),
+		c.HardPodAffinitySymmetricWeight,
+	)
+
+	if _, err := os.Stat(c.SchedulerConfigFile); err == nil {
+		//configData, err = ioutil.ReadFile(c.SchedulerServer.PolicyConfigFile)
+		configData, err = ioutil.ReadFile(c.PolicyConfigFile)
+		if err != nil {
+			return false, fmt.Errorf("unable to read scheduler config: %v", err)
```

Comment:
This was `true, error`.  They had a file, we can't read it, intent is not fulfilled.  We should die early.